src/azure-cli/azure/cli/command_modules/hdinsight/_client_factory.py

```diff
@@ -25,15 +25,8 @@ def cf_network(cli_ctx, aux_subscriptions=None, **_):
 
 
 def cf_graph(cli_ctx, **_):
-    from azure.cli.core._profile import Profile
-    from azure.cli.core.commands.client_factory import configure_common_settings
-    from azure.graphrbac import GraphRbacManagementClient
-    profile = Profile(cli_ctx=cli_ctx)
-    cred, _, tenant_id = profile.get_login_credentials(
-        resource=cli_ctx.cloud.endpoints.active_directory_graph_resource_id)
-    client = GraphRbacManagementClient(cred, tenant_id,
-                                       base_url=cli_ctx.cloud.endpoints.active_directory_graph_resource_id)
-    configure_common_settings(cli_ctx, client)
+    from azure.cli.command_modules.role import graph_client_factory
+    client = graph_client_factory(cli_ctx)
```

Comment:
I am glad to see the new `azure.cli.command_modules.role.graph_client_factory` interface significantly reduces the effort to build a Graph client. 😊

Reply:
> I am glad to see the new `azure.cli.command_modules.role.graph_client_factory` interface significantly reduces the effort to build a Graph client. 😊

learnt this from key vault PR 😁